sky/spot/controller.py

```diff
@@ -86,8 +86,21 @@ def _run(self):
                                                    self._cluster_name)
 
             if job_status is not None and not job_status.is_terminal():
-                # The job is healthy, continue to monitor the job status.
-                continue
+                need_recovery = False
+                if self._task.num_nodes > 1:
+                    # Check the cluster status for multi-node jobs, since the
+                    # job may not FAILED immediately when part of the nodes are
+                    # preempted.
+                    (cluster_status,
+                     handle) = backend_utils.refresh_cluster_status_handle(
+                         self._cluster_name, force_refresh=True)
+                    if cluster_status != global_user_state.ClusterStatus.UP:
+                        # recover the cluster if it is not up.
```

Comment:
Consider printing a message of the current status for better debugging?

Reply:
Good point! Added. Thanks!